plugin/pulp/plugin/distributor.py

```diff
@@ -0,0 +1,74 @@
+from pulp.platform.models import Distributor as PlatformDistributor
+
+
+class Distributor(PlatformDistributor):
+    """
+    The base Distributor object to publish content
+
+    This is meant to be subclassed by plugin authors as an opportunity to provide:
+
+    * Plugin specific publish functionality
+    * Add persistent data attributes for a plugin distributor subclass
+    * Additional validation of plugin distributor subclass configuration
+
+    The publish implementation is provided by :meth: `Distributor.publish` which provides more
+    details. Failing to implement this method will prevent publish functionality for this plugin
+    type.
+
+    This object is a Django model that inherits from :class: `pulp.platform.models.Distributor`
+    which provides the platform persistent attributes for a distributor object. Plugin authors can
+    add additional persistent distributor data by subclassesing this object and adding Django
+    fields. We defer to the Django docs on extending this model definition with additional fields.
+
+    Validation is done the Django way, so custom validation can also be added the same as any
+    Django model. We defer to the Django docs on adding custom validation to the subclassed
+    distributor. If any of the model validation methods are overridden, be sure to call super() so
+    the platform can still perform its validation.
+
+    Instantiation and calling of the subclassed plugin Distributor is described in detail in the
+    :meth: `Distributor.publish` method.
+    """
+
+    def publish(self):
+        """
+        Perform a publish
+
+        It is expected that plugins wanting to support publish will provide an implementation on the
+        subclassed Distributor.
+
+        The publish to perform uses the model attributes to encapsulate all of the information
```

Comment:
I find the first few words a little awkward to read. I think you could just cut them out and reduce this to "The model attributes encapsulate all of the information required."